### PR TITLE
Issue #3829: Test all projects for `type =` in their .info file

### PR DESCRIPTION
**Pull request by klonos**
https://github.com/backdrop/backdrop-issues/issues/3829

### CODE REVIEW

File: core/modules/system/tests/system.test [hunk 1]
```diff
@@ -1946,81 +1946,30 @@ array_space[a b] = Value';
...
 class InfoFileCoreTest extends BackdropUnitTestCase {
   /**
-   * Tests that all core modules' .info files have the project type string set to "module".
+   * Tests that the .info files of all core projects specify a project type.
    */
-  function testModuleInfoFileContents() {
-    $modules = backdrop_system_listing('/^' . BACKDROP_PHP_FUNCTION_PATTERN . '\.module$/', 'core/modules', 'name', 0);
+  function testProjectInfoFileContents() {
+    $projects = backdrop_system_listing('/^' . BACKDROP_PHP_FUNCTION_PATTERN . '\.info$/', 'core');
+    $allowed_project_types = array('module', 'theme', 'layout', 'profile', 'test');
```
Comment (quicksketch) [inline core/modules/system/tests/system.test]:
```suggestion
    $allowed_project_types = array('module', 'theme', 'layout', 'profile');
```

**Comment by klonos** [comment]
...making sure we get a fresh test run, after multiple commits

**Comment by klonos** [comment]
...OK, both php53/php70 tests have failed after the intentional changes; as expected:

Screen Shot 2019-06-01 at 9 45 42 am
Screen Shot 2019-06-01 at 9 46 21 am

File: core/modules/simpletest/tests/themes/test_basetheme/test_basetheme.info [hunk 2]
```diff
@@ -1,7 +1,8 @@
 name = Theme test base theme
 description = Test theme which acts as a base theme for other test subthemes.
+version = BACKDROP_VERSION
 backdrop = 1.x
-type = theme
+type = theme_intentionally_wrong
```
Comment (klonos) [inline core/modules/simpletest/tests/themes/test_basetheme/test_basetheme.info]:
```suggestion
type = theme
```

**Comment by quicksketch** [comment]
Nice work @klonos.

**Comment by quicksketch** [comment]
Thanks! Merged into 1.x and 1.13.x.

**Pull request merged**